File: frontend/src/App/Model/Match/Entities/Round.ts
import type { Side } from "../Types/Side";
import type { RoundWin } from "./RoundWin";
import type { KillEvent } from "./KillEvent";
import type { BombEvent } from "./BombEvent";

export interface Round {
  number: number;
  startedAt: string;
  endedAt: string;
  duration: number;
  winner: RoundWin;
  playerSideBySteamId: Record<string, Side>;
  kills: KillEvent[];
  bombEvents: BombEvent[];
}

File: backend/API/Models/Round.cs
using API.Models.Enums;

namespace API.Models;

public class Round
{
    public int Number { get; set; }
    public DateTime StartedAt { get; set; }
    public DateTime EndedAt { get; set; }
    public TimeSpan Duration => EndedAt - StartedAt;
    public RoundWin Winner { get; set; }
    public Dictionary<string, Side> PlayerSideBySteamId { get; set; }
    public List<KillEvent> Kills { get; set; } = [];
    public List<BombEvent> BombEvents { get; set; } = [];
}